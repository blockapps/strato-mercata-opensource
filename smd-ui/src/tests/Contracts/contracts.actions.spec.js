import {
  fetchContracts,
  fetchContractsSuccess,
  fetchContractsFailure,
  changeContractFilter
} from '../../components/Contracts/contracts.actions';
import { contracts, filter, error } from './contractsMock';

describe('Contracts: action', () => {

  test('change contract filter', () => {
    expect(changeContractFilter(filter)).toMatchSnapshot();
  });

  describe('fetch contracts', () => {

    test('request', () => {
      let chainId = "ff7ef45acb7a775018bc765b6fdeea432aaddfcd846cf6dd9442724266b1eac9";
      const limit = 10;
      const offset = 0;
      expect(fetchContracts(chainId, limit, offset)).toMatchSnapshot();
    });

    test('success', () => {
      expect(fetchContractsSuccess(contracts)).toMatchSnapshot();
    });

    test('failure', () => {
      expect(fetchContractsFailure(error)).toMatchSnapshot();
    });

  })

});