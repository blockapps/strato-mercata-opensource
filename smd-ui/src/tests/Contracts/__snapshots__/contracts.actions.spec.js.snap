// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contracts: action change contract filter 1`] = `
Object {
  "filter": "Just to test filter",
  "type": "CHANGE_CONTRACT_FILTER",
}
`;

exports[`Contracts: action fetch contracts failure 1`] = `
Object {
  "error": "ERROR",
  "type": "FETCH_CONTRACTS_FAILED",
}
`;

exports[`Contracts: action fetch contracts request 1`] = `
Object {
  "chainId": "ff7ef45acb7a775018bc765b6fdeea432aaddfcd846cf6dd9442724266b1eac9",
  "limit": 10,
  "name": undefined,
  "offset": 0,
  "type": "FETCH_CONTRACTS",
}
`;

exports[`Contracts: action fetch contracts success 1`] = `
Object {
  "contracts": Object {
    "Cloner": Array [
      Object {
        "address": "d07e932212f7f368b6948ffd96e1d4c726c8395d",
        "createdAt": 1512480630000,
      },
      Object {
        "address": "2c6619f0418c2f191e2225091f7692363a91c336",
        "createdAt": 1512480770000,
      },
    ],
    "GreeterA": Array [
      Object {
        "address": "0293f9b10a4453667db7fcfe74728c9d821add4b",
        "createdAt": 1512481078000,
      },
    ],
    "GreeterB": Array [
      Object {
        "address": "GreeterA",
        "createdAt": 1512481078000,
      },
    ],
  },
  "type": "FETCH_CONTRACTS_SUCCESSFUL",
}
`;
